source/pipeline/extract.py

```diff
@@ -0,0 +1,169 @@
+"""Script which gets data from web api search console, one game at a time"""
+
+import argparse
+import re
+import json
+import os
+from bs4 import BeautifulSoup
+from forex_python.converter import CurrencyRates
+import requests
+
+
+FOLDER_PATH = 'data/'  # needs to be /tmp/data for lambda
+
+STEAM_PATH = f'{FOLDER_PATH}steam_products.json'
+STEAM_SEARCH = 'https://store.steampowered.com/search?term={search_term}'
+STEAM_SPLIT = '<div class="search_name ellipsis">'
+
+GOG_PATH = f'{FOLDER_PATH}gog_products.json'
+GOG_SEARCH = 'https://catalog.gog.com/v1/catalog?limit=48&query=like%3A{search_term}'
+DEFAULT_RATE = 0.77
+
+# <--- Steam functions --->
+
+
+def get_steam_html(search_input: str) -> str:
+    """Get first result data from steam search term"""
+    # build url
+    response = requests.get(STEAM_SEARCH.format(search_term=search_input))
+    raw_data = response.text
+    raw_data = raw_data.split(STEAM_SPLIT)
+    if len(raw_data) <= 1:
+        print(f'{search_input} is invalid and leads to no match for Steam')
+        return ''  # len = 1 means no search results found, return falsey value
+    # else get the first result ignoring the headers
+    return raw_data[1]
+
+
+def parse_steam(data: str, search_input: str) -> dict:
+    """Function to scrape top selling games and output list of dicts with prices and titles"""
+    soup = BeautifulSoup(data, 'html.parser')
+
+    # class = "discount_original_price" >£10.99 # if none, then no discount
+    # class = "discount_final_price" >£10.99
+
+    title = soup.find("span", {"class": "title"})
+    if title:
+        title = title.get_text().strip()
+    else:
+        return {}  # if no match
```

Comment:
Suggestion: use a guard clause here

```suggestion
    if not title:
        return {}  # if no match
        
    title = title.get_text().strip()
```